rust/ql/lib/codeql/rust/elements/internal/StmtListImpl.qll

```diff
@@ -12,17 +11,40 @@ private import codeql.rust.elements.internal.generated.StmtList
  * be referenced directly.
  */
 module Impl {
+  // the following QLdoc is generated: if you need to edit it, do it in the schema file
   /**
-   * A list of statements in a block.
+   * A list of statements in a block, with an optional tail expression at the
+   * end that determines the block's value.
    *
    * For example:
    * ```rust
    * {
    *     let x = 1;
    *     let y = 2;
+   *     x + y
    * }
    * //  ^^^^^^^^^
    * ```
    */
-  class StmtList extends Generated::StmtList { }
+  class StmtList extends Generated::StmtList {
+    /**
+     * Gets the `index`th statement or expression of this statement list (0-based).
+     */
+    AstNode getStmtOrExpr(int index) {
```

Comment:
What is the use case for this compared to using `getStatement` and `getTailExpr`? Could it be creating more confusion than it solves given that something like `{ foo(); bar() }` will return a statement expression for `0` and an expression directly for `1`?